refactor(formatter): improve the code that was made for removing the binding pattern

Dunqing opened the pull request:
Follow up on https://github.com/oxc-project/oxc/pull/15925. Improved slightly to make it simpler.

camc314 reviewed (approved):
thank you 🙏

leaysgur reviewed (approved):
Oh, Prettier does not have tests for `!`... 👀

Dunqing committed:
refactor(formatter): improve the code that was made for removing the binding pattern (#17277)

Follow up on https://github.com/oxc-project/oxc/pull/15925. Improved slightly to make it simpler.

Dunqing commented:
> Oh, Prettier does not have tests for `!`... 👀

Yes, it's surprised me, and I found another regression from https://github.com/oxc-project/oxc-ecosystem-ci/actions/runs/20449000316/job/58758212726, which fixed in https://github.com/oxc-project/oxc/pull/17277/commits/28ee9a8d9b2db0a59a878231975d3316eb30cb89
